addons/buildables/functions/fnc_areaBuild.sqf

```diff
@@ -0,0 +1,41 @@
+#include "script_component.hpp"
+/*
+ * Author: veteran29
+ * Build area objects.
+ *
+ * Arguments:
+ * 0: Area hash <HASHMAP>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+ */
+
+params [
+    ["_area", createHashMap]
+];
+
+private _objects = getMissionLayerEntities (_area get "layer") select 0;
+private _buildTime = _area get "buildTime";
+
+if (_buildTime < 1) exitWith {
+    {_x hideObjectGlobal false} forEach _objects;
+
+    [QGVAR(built), _area get "layer"] call CBA_fnc_globalEvent;
+};
+
+private _delay = _buildTime / count _objects;
+
+private _fnc_build = {
+    params ["_fnc_build", "_delay", "_objects", "_area"];
+    if (_objects isEqualTo []) exitWith {
+        [QGVAR(built), _area get "layer"] call CBA_fnc_globalEvent;
+    };
+
+    (_objects deleteAt 0) hideObjectGlobal false;
```

Comment:
We agreed it can be done better like this:
```suggestion
    ([_objects deleteAt 0] param [0, objNull]) hideObjectGlobal false;
    
    if (_objects isEqualTo []) exitWith {
        [QGVAR(built), _area get "layer"] call CBA_fnc_globalEvent;
    };
```
This way we end the script right after last object appears, not a loop later.